sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py

```diff
@@ -3,362 +3,233 @@
+            days=generated.days,
+        )
 
 
-class FileSystemPropertiesPaged(ContainerPropertiesPaged):
-    """An Iterable of File System properties.
+class Metrics(GenMetrics):
+    """A summary of request statistics grouped by API in hour or minute aggregates.
 
-    :ivar str service_endpoint: The service URL.
-    :ivar str prefix: A file system name prefix being used to filter the list.
-    :ivar str marker: The continuation token of the current page of results.
-    :ivar int results_per_page: The maximum number of results retrieved per API call.
-    :ivar str continuation_token: The continuation token to retrieve the next page of results.
-    :ivar str location_mode: The location mode being used to list results. The available
-        options include "primary" and "secondary".
-    :ivar current_page: The current page of listed results.
-    :vartype current_page: list(~azure.storage.filedatalake.FileSystemProperties)
+    All required parameters must be populated in order to send to Azure.
 
-    :param callable command: Function to retrieve the next page of items.
-    :param str prefix: Filters the results to return only file systems whose names
-        begin with the specified prefix.
-    :param int results_per_page: The maximum number of file system names to retrieve per
-        call.
-    :param str continuation_token: An opaque continuation token.
+    :keyword str version:
+        The version of Storage Analytics to configure. The default value is 1.0.
+    :keyword bool enabled:
+        Indicates whether metrics are enabled for the Datalake service.
+        The default value is `False`.
+    :keyword bool include_apis:
+        Indicates whether metrics should generate summary statistics for called API operations.
+    :keyword ~azure.storage.filedatalake.RetentionPolicy retention_policy:
+        Determines how long the associated data should persist. If not specified the retention
+        policy will be disabled by default.
     """
 
-    def __init__(self, *args, **kwargs):
-        super(FileSystemPropertiesPaged, self).__init__(
-            *args,
-            **kwargs
-        )
-
-    @staticmethod
-    def _build_item(item):
-        return FileSystemProperties._from_generated(item)  # pylint: disable=protected-access
+    version: str = '1.0'
+    """The version of Storage Analytics to configure."""
+    enabled: bool = False
+    """Indicates whether metrics are enabled for the Datalake service."""
+    include_apis: bool
+    """Indicates whether metrics should generate summary statistics for called API operations."""
+    retention_policy: RetentionPolicy = RetentionPolicy()
+    """Determines how long the associated data should persist."""
 
+    def __init__(self, **kwargs: Any) -> None:
+        self.version = kwargs.get('version', '1.0')
+        self.enabled = kwargs.get('enabled', False)
+        self.include_apis = kwargs.get('include_apis')  # type: ignore [assignment]
```

Comment:
Do we need this ignore? I think the docstring may have been inaccurate here and `include_apis` should be `Optional[bool]`. See other packages to verify this (i.e. Queue or Blobs?)

Reply:
Thanks for verifying this! It turns out in the tests and other packages, `include_apis` can be `None`. Will adjust this according to the feedback.